packages/itwinui-react/src/core/Table/types/react-table-types.d.ts

```diff
@@ -0,0 +1,1320 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/* eslint-disable @typescript-eslint/no-empty-interface */
```

Comment:
Is this file copied from react-table? Maybe it's worth making a note that we are copying it and for what reasons.

Reply:
Yes, it is copied from `@types/react-table`. Will be adding clearer comments (https://github.com/iTwin/iTwinUI/pull/1514#discussion_r1316007264)

Reply:
> Maybe it's worth making a note that we are copying it and for what reasons.

https://github.com/iTwin/iTwinUI/pull/1514/commits/0f47a7a6f6b7f85f63c0e9e057f2195d2a2f0a91: Added a few reasons